### PR TITLE
Julia: add mentoring notes for clock

### PR DESCRIPTION
**Pull request by cmcaine**
Code examples are lightly edited by me. Errors are my own, etc.

@ScottPJones, I edited your code slightly to remove an `==` method and
to use `value()` rather than `.value`. Let me know if you think your
original version should stand.

Also let me know if you'd prefer I not use your code as an example :)

### CODE REVIEW

File: tracks/julia/exercises/clock/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+## Things to watch out for
+
+- It is preferred to use \``value(m)`\` rather than \``m.value`\`. Fields are generally considered private in Julia. You can import \``value`\` from Dates.
```
Comment (SaschaMann) [inline tracks/julia/exercises/clock/mentoring.md:3]: I think an explanation of why and when this is the case would be helpful. On first look, this sounds like one has to add Java levels of boilerplate without (m)any benefits.

File: tracks/julia/exercises/clock/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    minutes::Int8
+    Clock(h,m) = new(rem(h + div(m,60,RoundDown),24,RoundDown),rem(m,60,RoundDown))
+end
+
+Base.:+(c::Clock,m::Dates.Minute) = Clock(c.hours, c.minutes + value(m))
+Base.:-(c::Clock,m::Dates.Minute) = Clock(c.hours, c.minutes - value(m))
+Base.:+(c::Clock,h::Dates.Hour) = Clock(c.hours + value(h), c.minutes)
+Base.:-(c::Clock,h::Dates.Hour) = Clock(c.hours - value(h), c.minutes)
+
+Base.show(io::IO, c::Clock) = print(io, '"', lpad(c.hours,2,'0'), ":", lpad(c.minutes,2,'0'), '"')
```
Comment (SaschaMann) [inline tracks/julia/exercises/clock/mentoring.md:116]: Perhaps add a suggestion above that Printf is a nicer solution to this than lpad.

File: tracks/julia/exercises/clock/mentoring.md (hunk 1 above)
Comment (cmcaine) [reply tracks/julia/exercises/clock/mentoring.md:3]: That's fair, but I'm not sure I have a good explanation beyond the same reasons that you'd do it for Java or any other language. Can you think of one?

File: tracks/julia/exercises/clock/mentoring.md (hunk 2 above)
Comment (cmcaine) [reply tracks/julia/exercises/clock/mentoring.md:116]: I'm not sure I want to venture that opinion. I personally prefer it, but I think Scott prefers this with lpad. printf is its own weird little language.

File: tracks/julia/exercises/clock/mentoring.md (hunk 1 above)
Comment (SaschaMann) [reply tracks/julia/exercises/clock/mentoring.md:3]: In some cases it makes the code more generic. `value(x)` will work for any x as long as there's a method for it, while `x.value` will require the field. You won't reap the benefits of multiple dispatch if you always access fields directly.

File: tracks/julia/exercises/clock/mentoring.md (hunk 2 above)
Comment (SaschaMann) [reply tracks/julia/exercises/clock/mentoring.md:116]: Maybe "if you're already familiar with Printf in other languages, consider using..." or "consider taking a look at Printf as a way to format values in a string, it's more complex but also more flexible than lpad" as an optional suggestion/talking point?

File: tracks/julia/exercises/clock/mentoring.md (hunk 1 above)
Comment (cmcaine) [reply tracks/julia/exercises/clock/mentoring.md:3]: I feel like that's the same benefit as in Java :shrug: But we could restate it, I suppose for the scientists and novices who might not be familiar.

**Comment by cmcaine** [comment]
I think both review comments are now addressed.

**Pull request merged**